components/pdf_preview.jsx

```diff
@@ -147,7 +146,10 @@ export default class PDFPreview extends React.PureComponent {
     }
 
     getPdfDocument = () => {
-        PDFJS.getDocument(this.props.fileUrl).then(this.onDocumentLoad).catch(this.onDocumentLoadError);
+        import('pdfjs-dist').then((PDFJS) => {
```

Comment:
This looks great to me, i only have one small convern however that the function `getPdfDocument` is also being called in [componentDidUpdate](https://github.com/mattermost/mattermost-webapp/blob/669f0aa9f07b3d803461b2ed3be187e84317375f/components/pdf_preview.jsx#L78) do you think it is fine to import the library once again on every update or can we store the copy of it in ref or something? just curious

Reply:
AFAIK that is handled by webpack, the first import load the module and insert in the list of existing modules, after that the module is reuse on every import.

Reply:
It does look a bit weird to be doing this setup of setting `disableWorker` repeatedly, but I don't think calling it multiple times should cause a problem. We do that in the `LatexBlock` component with `katex`, and even though that's in `componentDidMount`, it'll still be called repeatedly when more of those are mounted and unmounted.

We could save the library in `this.state.pdfjs` like we do in `LatexBlock` for consistency, but I don't think it'd make a functional difference

Reply:
OK if you both feel like re importing doesn't make any difference whatsoever i am good with it 👍🏽

Reply:
Yes,  I think this is clean enough, adding state when is not needed doesn't looks like very useful.

Reply:
If I were to do it I wouldn't have added state for it but ref would have been fine. With out both is also fine